src/server/ua_services_attribute.c

```diff
@@ -108,15 +108,18 @@ compatibleValueRankArrayDimensions(UA_Int32 valueRank, size_t arrayDimensionsSiz
             return UA_STATUSCODE_BADTYPEMISMATCH;
         break;
     case 0: /* the value is an array with one or more dimensions */
-        if(arrayDimensionsSize < 1)
+        if(arrayDimensionsSize > 3)
```

Comment:
I'm not sure about that change.
For the valueRank 0, scalars are not permitted. This check would allow them.
Also, is there a reason to not allow multi-arrays of more than three dimensions?